src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -126,6 +134,89 @@ public void deleteSession(Session target) {
         addressBook.removeSession(target);
     }
 
+    /**
+     * Assigns the student/tutor to a session in addressBook
+     * @param assignment
+     */
+    public void registerAssignment(Assignment assignment) throws TimeClashException, AlreadyEnrolledException,
```

Comment:
I think need to abstract the method here. You can move the null checks to the assignment class and assignment command parser to comply with the other model objects/methods. Since the Model and model manager are mostly facades.